client/fed_test.go

```diff
@@ -505,3 +505,78 @@ func TestStatHttp(t *testing.T) {
 		assert.Contains(t, err.Error(), "Do not understand the destination scheme: some. Permitted values are file, osdf, pelican, stash, ")
 	})
 }
+
+// Test the functionality of the direct reads feature (?DirectRead=true)
+func TestDirectReads(t *testing.T) {
+	t.Run("testDirectReadsSuccess", func(t *testing.T) {
+		viper.Reset()
+		server_utils.ResetOriginExports()
+		viper.Set("Origin.EnableDirectReads", true)
+		fed := fed_test_utils.NewFedTest(t, bothPublicOriginCfg)
+		for _, export := range fed.Exports {
+			testFileContent := "test file content"
+			// Drop the testFileContent into the origin directory
+			tempFile, err := os.Create(filepath.Join(export.StoragePrefix, "test.txt"))
+			assert.NoError(t, err, "Error creating temp file")
+			defer os.Remove(tempFile.Name())
+			_, err = tempFile.WriteString(testFileContent)
+			assert.NoError(t, err, "Error writing to temp file")
+			tempFile.Close()
+
+			viper.Set("Logging.DisableProgressBars", true)
+
+			// Set path for object to upload/download
+			tempPath := tempFile.Name()
+			fileName := filepath.Base(tempPath)
+			uploadURL := fmt.Sprintf("pelican://%s:%s%s/%s?directread", param.Server_Hostname.GetString(), strconv.Itoa(param.Server_WebPort.GetInt()),
+				export.FederationPrefix, fileName)
+
+			// Download the file with GET. Shouldn't need a token to succeed
+			transferResults, err := client.DoGet(fed.Ctx, uploadURL, t.TempDir(), false)
+			assert.NoError(t, err)
+			if err == nil {
```

Comment:
Here we do because even if assert fails for assert.NoError, we still move on to the assert.Equal after. This is bad because if we fail, then transferResults is nil and we get a nil pointer dereference. That is why we want to do the if err == nil here

Reply:
Maybe do a `require.NoError` instead?